x/staking/types/keys.go

```diff
@@ -378,3 +381,29 @@ func GetREDsByDelToValDstIndexKey(delAddr sdk.AccAddress, valDstAddr sdk.ValAddr
 func GetHistoricalInfoKey(height int64) []byte {
 	return append(HistoricalInfoKey, []byte(strconv.FormatInt(height, 10))...)
 }
+
+func GetValidatorConsPubKeyRotationHistoryKey(history ConsPubKeyRotationHistory) []byte {
+	return append(append(
+		ValidatorConsPubKeyRotationHistoryKey,
+		[]byte(history.OperatorAddress)...),
+		sdk.Uint64ToBigEndian(uint64(history.Height))...)
+}
+
+func GetBlockConsPubKeyRotationHistoryKey(history ConsPubKeyRotationHistory) []byte {
+	return append(append(
+		BlockConsPubKeyRotationHistoryKey,
+		sdk.Uint64ToBigEndian(uint64(history.Height))...),
+		[]byte(history.OperatorAddress)...)
+}
+
+func GetValdiatorConsPubKeyRotationHistoryPrefix(operatorAddress string) []byte {
```

Comment:
nit:  typo on validator
```suggestion
func GetValidatorConsPubKeyRotationHistoryPrefix(operatorAddress string) []byte {
```

Reply:
Thanks for the correction